### PR TITLE
Add some bounds checks

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

**Comment by janisozaur** [comment]
[Canyoneer.by.Xophe_crash.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/1700913/Canyoneer.by.Xophe_crash.sv6.zip)

**Comment by IntelOrca** [comment]
Merging this PR would be a disaster. Assertions pop up for many of the saved games I load.

**Comment by IntelOrca** [comment]
Here is one of the places that needs fixing... it needs to check if ride type is `RIDE_TYPE_NULL` before calling `ride_type_set_invented`.

https://github.com/OpenRCT2/OpenRCT2/blob/214df1cf75535f8d9beef217383af1c17b59b5c8/src/openrct2/management/Research.cpp#L939-L942

File: src/openrct2/management/Research.cpp [hunk 1]
```diff
@@ -938,7 +940,11 @@ void research_fix()
 
                 for (uint8 j = 0; j < MAX_RIDE_TYPES_PER_RIDE_ENTRY; j++)
                 {
-                    ride_type_set_invented(rideEntry->ride_type[j]);
+                    uint32 rideType = rideEntry->ride_type[j];
+                    if (rideType != RIDE_TYPE_NULL)
```
Comment (Gymnasiast) [inline src/openrct2/management/Research.cpp]: There's little point in checking for this.

Comment (janisozaur) [reply src/openrct2/management/Research.cpp]: Why?

Comment (Gymnasiast) [reply src/openrct2/management/Research.cpp]: Oh wait, scratch that. I made that array the size of RIDE_TYPE_COUNT. Sorry, please ignore what I said.

**Comment by IntelOrca** [comment]
It is probably okay now, but if assertions start popping up which didn't before then players will complain even if wasn't right - it can make a playable game unplayable.

Do you intend to investigate the reason the assert is called? @janisozaur

**Comment by Gymnasiast** [comment]
Certainly, but then it previously wrote/checked out of bounds, which can lead to far more harder to investigate breakage than an assertion being hit. After all, that's why they exist in the first place.

**Comment by janisozaur** [comment]
@IntelOrca wrt `peep_update_ride_prepare_for_state_9`: I checked, in the case of attached park it gets imported this way from sv6.

**Comment by IntelOrca** [comment]
We have to do something sensible. Perhaps fallback to first valid station exit and if that fails, delete the peep or something.

**Comment by janisozaur** [comment]
@IntelOrca done + rebased

**Comment by duncanspumpkin** [comment]
A while back there was a bug that caused the peeps to end up with invalid stations.  That might be related to this.

**Comment by IntelOrca** [comment]
@janisozaur that isn't what I had in mind. I meant rather than assert which just crashes the game (if state ends up like that during gameplay), change the peep exit. Also it needs to be changed to a valid station with an exit, that might not necessarily be station 0.

**Comment by janisozaur** [comment]
@IntelOrca I don't quite agree with silently fixing this. If such a situation is to happen in runtime, I would gladly accept a bug report for it and with fixing it upon import we still allow to continue playing even on malformed sv6 like the one attached.

**Comment by janisozaur** [comment]
@duncanspumpkin wouldn't that bug be because of https://github.com/OpenRCT2/OpenRCT2/commit/5db0a05f11cb0f7c7df6303263ffc5e820290df7 ? It selects a potentially unavailable station, just like I did a commit before. I've chosen to remove such guests

**Comment by IntelOrca** [comment]
@jaanisozaur would a warning be acceptable if we fix it at runtime?

**Comment by janisozaur** [comment]
> We should always guard against null deref

https://github.com/OpenRCT2/OpenRCT2/pull/6975

**Comment by Gymnasiast** [comment]
I think this was merged too soon. And I don't think we should remove peeps with an invalid station, but simply reset their interaction ride to RIDE_ID_NULL.

**Pull request merged**